prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -328,29 +332,49 @@ class FHIRRouter(
             // JURIS FILTER
             //  default: allowNone
             var passes = evaluateFilterCondition(getJurisFilters(receiver, orgFilters), bundle, false)
+
             // QUALITY FILTER
-            //  default: must have message id, patient last name, patient first name, dob, specimen type
-            //           must have at least one of patient street, zip code, phone number, email
-            //           must have at least one of order test date, specimen collection date/time, test result date
-            passes = passes &&
-                evaluateFilterCondition(
-                    getQualityFilters(receiver, orgFilters),
+            if (passes) {
+                //  default: must have message id, patient last name, patient first name, dob, specimen type
+                //           must have at least one of patient street, zip code, phone number, email
+                //           must have at least one of order test date, specimen collection date/time, test result date
+                val filters = getQualityFilters(receiver, orgFilters)
```

Comment:
Remove this val and put this code online.  Kotlin is about reducing unneeded lines of code.

Reply:
the val is used twice, but with the other changes I put in the code change is not much